src/harness/reference_models/interference/interference.py

```diff
@@ -0,0 +1,676 @@
+  PPA_AREA = 2
+  FSS_CO_CHANNEL = 3
+  FSS_BLOCKING = 4
+  ESC = 5
+
+# Global container to store neighborhood distance type of all the protection 
+_DISTANCE_PER_PROTECTION_TYPE = {
+   ProtectedEntityType.GWPZ_AREA :  (GWPZ_NEIGHBORHOOD_DIST, GWPZ_NEIGHBORHOOD_DIST),
+   ProtectedEntityType.PPA_AREA : ( PPA_NEIGHBORHOOD_DIST,  PPA_NEIGHBORHOOD_DIST),
+   ProtectedEntityType.FSS_CO_CHANNEL : ( FSS_CO_CHANNEL_NEIGHBORHOOD_DIST,  FSS_CO_CHANNEL_NEIGHBORHOOD_DIST),
+   ProtectedEntityType.FSS_BLOCKING : ( FSS_BLOCKING_NEIGHBORHOOD_DIST,  FSS_BLOCKING_NEIGHBORHOOD_DIST),
+   ProtectedEntityType.ESC: (ESC_NEIGHBORHOOD_DIST_A, ESC_NEIGHBORHOOD_DIST_B)
+    }
+
+# Define CBSD grant, i.e., a tuple with named fields of 'latitude',
+# 'longitude', 'height_agl', 'indoor_deployment', 'antenna_azimuth',
+# 'antenna_gain', 'antenna_beamwidth', 'cbsd_category', 
+# 'max_eirp', 'low_frequency', 'high_frequency', 'is_managed_grant'
+CbsdGrantInformation = namedtuple('CbsdGrantInformation',
+                       ['latitude', 'longitude', 'height_agl', 
+                        'indoor_deployment', 'antenna_azimuth', 'antenna_gain',
+                        'antenna_beamwidth', 'cbsd_category', 
+                        'max_eirp', 'low_frequency', 'high_frequency', 
+                        'is_managed_grant'])
+
+# Define protection constraint, i.e., a tuple with named fields of
+# 'latitude', 'longitude', 'low_frequency', 'high_frequency'
+ProtectionConstraint = namedtuple('ProtectionConstraint',
+                                  ['latitude', 'longitude', 'low_frequency',
+                                   'high_frequency', 'entity_type'])
+
+# Define FSS Protection Point, i.e., a tuple with named fields of
+# 'latitude', 'longitude', 'height_agl', 'max_gain_dbi', 'pointing_azimuth',
+# 'pointing_elevation' 
+FssProtectionPoint = namedtuple('FssProtectionPoint',
+                                   ['latitude', 'longitude',
+                                    'height_agl', 'max_gain_dbi',
+                                    'pointing_azimuth', 'pointing_elevation'])
+
+
+# Define ESC information, i.e., a tuple with named fields of
+# 'antenna_height', 'antenna_azimuth', 'antenna_gain', 'antenna_pattern_gain'
+EscInformation = namedtuple('EscInformation',
+                                 ['antenna_height', 'antenna_azimuth',
+                                  'antenna_gain', 'antenna_pattern_gain'])
+
+class AggregateInterferenceOutputFormat:
+  """Implementation of output format of aggregate interference 
+
+  This class is used to generate specified output format for post IAP 
+  allowed interference or for aggregate interference to non-DPAs
+  
+  Creates a data structure to store post IAP allowed interference or 
+  aggregate interference to non-DPAs output in the format of 
+  {latitude : {longitude : [ output_first_5MHz_segment, output_first_5MHz_segment ]}}
+  """
+  def __init__(self):
+    self.manager = multiprocessing.Manager()
```

Comment:
You are using a Manager proxy object to be able to share the dict across process, so that I guess we fill it up in parallel.
That is a good idea in general, however I am not sure it will work in this case since we have a inner dict.
See for example: https://stackoverflow.com/questions/9436757/how-does-multiprocessing-manager-work-in-python

We need to do a test. But I guess we can keep this issue in mind, and fix it in a future PR so as not to block this PR.

Reply:
Test   is done on this. It is working as expected.    Below approach mentioned in the python document is used to achieve   this :           "If standard (non-proxy) list or dict objects are contained in a   referent, modifications to those mutable values will not be propagated   through the manager because the proxy has no way of knowing when the values   contained within are modified. However, storing a value in a container proxy   (which triggers a __setitem__ on the proxy object) does propagate through the   manager and so to effectively modify such an item, one could re-assign the   modified value to the container proxy:          # create a list proxy and append a mutable object (a dictionary)     lproxy = manager.list()     lproxy.append({})     # now mutate the dictionary     d = lproxy[0]     d['a'] = 1     d['b'] = 2     # at this point, the changes to d are not yet synced, but by     # updating the dictionary, the proxy is notified of the change     lproxy[0] = d "          Even though we are using proxy container, sometime output is going hay-wire   because of multiple proocesses. Multiprocessing.Lock is used to synchronize   writing to shared dictionary across multiple processes